desc/equilibrium/coords.py

```diff
@@ -376,58 +375,100 @@ def fixup(x, *args):
-            theta_PEST,
-            rho,
-            zeta,
-        )
+        theta, (res, niter) = theta
     out = jnp.column_stack([rho, jnp.atleast_1d(theta.squeeze()), zeta])
     if full_output:
         return out, (res, niter)
     return out
 
 
-# TODO(#568): decide later whether to assume given phi instead of zeta.
+def _partial_sum(L, L_lmn, W, W_lmn, iota):
+    """Convert FourierZernikeBasis to set of Fourier series.
+
+    TODO(#1243) Do proper partial summation once the DESC
+    basis are improved to store the padded tensor product modes.
+    https://github.com/PlasmaControl/DESC/issues/1243#issuecomment-3131182128.
+
+    Parameters
+    ----------
+    L : Transform
+        FourierZernikeBasis
+    L_lmn : jnp.ndarray
+        FourierZernikeBasis basis coefficients for λ.
+    W : Transform
+        FourierZernikeBasis
+    W_lmn : jnp.ndarray
+        FourierZernikeBasis basis coefficients for ω.
+    iota : jnp.ndarray
+        Shape (L.grid.num_rho, )
+
+    Returns
+    -------
+    lmbda_minus_iota_omega, modes
+        Spectral coefficients and modes.
+        Shape (num rho, num zeta, num modes).
+
+    """
+    grid = L.grid
+    errorif(not grid.fft_poloidal, NotImplementedError, msg="See note in docstring.")
+    # TODO: (#568)
+    warnif(
+        grid.M > L.basis.M,
+        ResolutionWarning,
+        msg="Poloidal grid resolution is higher than necessary for coordinate mapping.",
+    )
+    warnif(
+        grid.M < L.basis.M,
+        ResolutionWarning,
+        msg="High frequency lambda modes will be truncated in coordinate mapping.",
+    )
+    lmbda_minus_iota_omega = L.transform(L_lmn)
+    lmbda_minus_iota_omega = (
+        rfft(grid.meshgrid_reshape(lmbda_minus_iota_omega, "rzt"), norm="forward")
+        .at[..., (0, -1) if ((grid.num_theta % 2) == 0) else 0]
+        .divide(2)
+        * 2
+    )
+    return lmbda_minus_iota_omega, jnp.fft.rfftfreq(grid.num_theta, 1 / grid.num_theta)
+
+
 def _map_clebsch_coordinates(
-    coords,
     iota,
+    alpha,
+    zeta,
     L_lmn,
-    L_basis,
-    guess=None,
+    L,
+    theta0=None,
     period=np.inf,
     tol=1e-6,
     maxiter=30,
-    full_output=False,
     **kwargs,
 ):
     """Find θ for given Clebsch field line poloidal label α.
 
     Parameters
     ----------
-    coords : ndarray
-        Shape (k, 3).
-        Clebsch field line coordinates [ρ, α, ζ]. Assumes ζ = ϕ.
-        Each row is a different point in space.
     iota : ndarray
-        Shape (k, ).
+        Shape (num iota, ).
         Rotational transform on each node.
+    alpha : ndarray
+        Shape (num alpha, )
+    zeta : ndarray
+        Shape (num zeta, )
     L_lmn : jnp.ndarray
-        Spectral coefficients for lambda.
-    L_basis : Basis
-        Spectral basis for lambda.
-    guess : jnp.ndarray
-        Shape (k, ).
+        Spectral coefficients for λ.
+    L : Transform
+        Transform for λ built on coordinates [ρ, θ, ζ].
```

Comment:
what is theta here?

also I'd vote to name this `L_transform` or something to be more explicit and not confuse with resolution parameter `L`

Reply:
DESC theta. This is currently a private method, eventually once 1243 is done, the passed in parameter will be changed to just the spectral coefficients (no transform) and we can then do proper partial summation

Reply:
I'll rename to what you want on `ku/nufft`. It's not worth running GitHub CI again for a name change for private function